masterfabric-mobile/swift-camp
# CL - Group Box Implementation

## 📋 PR Description

This PR introduces the GroupBoxView component, which displays dynamic content using a presenter. It includes a back button linked to the goBack action, a scrollable section featuring an image and a descriptive text, and an interactive "like" button with a toggling state. The design leverages SwiftUI's features to provide a clean and user-friendly interface while ensuring smooth user interaction.

---

## ✅ Checklist


---

## 🛠 Steps to Test

1.	Launch the app and navigate to the home screen.
2.	Open the GroupBox module from the home screen.
3.	Verify the GroupBox displays the correct title, description, and toggle.
4.	Test the toggle functionality and ensure the agreement state updates correctly.
5.	Navigate back to the home screen and confirm smooth transition.

## 🔗 Related Links

- Documentation: https://developer.apple.com/documentation/swiftui/groupbox

### 📷 Screenshots (Optional)

<table>
  <thead>
    <tr>
      <th style="padding-left: 20px;">GroupBox Page</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="padding-right: 20px;">
        <img src="https://github.com/user-attachments/assets/365bf67d-1f9e-4212-af02-ab126bcff3c4" alt="home page grouupbox" width="400"/>
      </td>
    </tr>
  </tbody>
</table>
